src/sync/app.cpp

```diff
@@ -73,10 +73,23 @@ App::App(const Config& config)
 
 static Optional<AppError> check_for_errors(const Response& response)
 {
+    auto http_status_code_is_fatal = [response] {
+        if (response.http_status_code >= 300
```

Comment:
This could just be `bool http_status_code_is_fatal = response.http_status_code >= 300 || || (response.http_status_code < 200 && response.http_status_code != 0)`; there's no real reason for it to be a function.

Reply:
True, I've cleaned this up